src/celeritas/ext/detail/ImportModelConverter.cc

```diff
@@ -0,0 +1,275 @@
+    auto iter = model_map.find(name);
+    if (iter == model_map.end())
+    {
+        static celeritas::TypeDemangler<G4VEmModel> demangle_model;
+        CELER_LOG(warning) << "Encountered unknown model '" << name
+                           << "' (RTTI: " << demangle_model(model) << ")";
+        return ImportModelClass::other;
+    }
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get a G4Material from a material index.
+ */
+G4Material const& get_g4material(unsigned int mat_idx)
+{
+    auto const* g4_cuts_table = G4ProductionCutsTable::GetProductionCutsTable();
+    CELER_EXPECT(mat_idx < g4_cuts_table->GetTableSize());
+    auto const* g4_material
+        = g4_cuts_table->GetMaterialCutsCouple(mat_idx)->GetMaterial();
+    CELER_ENSURE(g4_material);
+    return *g4_material;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Get a G4ParticleDefinition reference from a particle identifier.
+ */
+G4ParticleDefinition const& get_g4particle(PDGNumber pdg)
+{
+    CELER_EXPECT(pdg);
+    auto* particle
+        = G4ParticleTable::GetParticleTable()->FindParticle(pdg.get());
+    CELER_ENSURE(particle);
+    return *particle;
+}
+
+//---------------------------------------------------------------------------//
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with materials, primary, and secondary.
+ */
+ImportModelConverter::ImportModelConverter(VecMaterial const& materials,
+                                           PDGNumber particle,
+                                           PDGNumber secondary)
+    : materials_(materials), particle_(particle), secondary_(secondary)
+{
+    CELER_EXPECT(particle_);
+    g4particle_ = &get_g4particle(particle);
+    CELER_ENSURE(g4particle_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Convert a model.
+ *
+ * - Model energy limits are set based on model, particle, and cutoff values
+ * - If the model class requires elemental cross sections, a reduced-resolution
+ *   energy grid is calculated and cross sections are evaluated for those grid
+ *   points.
+ */
+ImportModel ImportModelConverter::operator()(G4VEmModel const& model) const
+{
+    ImportModel result;
+    result.model_class = to_import_model(model);
+
+    // Calculate lower cutoff energy for the model in each material
+    result.materials.resize(materials_.size());
+
+    for (auto mat_idx : celeritas::range(materials_.size()))
+    {
+        G4Material const& g4mat = get_g4material(mat_idx);
+
+        // Calculate lower and upper energy bounds
+        double min_energy
+            = std::max(model.LowEnergyLimit(),
+                       const_cast<G4VEmModel&>(model).MinPrimaryEnergy(
+                           &g4mat, g4particle_, this->get_cutoff(mat_idx)))
+              / MeV;
```

Comment:
I'm not sure that using these `ElementSelector` micro xs energy bounds for the model applicability will work in general. For example:
- The Geant4 `LowEnergyLimit()` for the photoelectric model is 0.1 keV. However, the photoelectric cross section is nonzero down to zero energy (in order to stop low energy photons), so the model should also be applicable down to zero.
- The lower energy bound for e+-->gg will also be 0.1 keV, but annihilation should be applicable down to zero.
- The lower energy bound for Rayleigh is 0.1 keV, but the Rayleigh macro xs is extrapolated outside the lambda table, so we could sample Rayleigh but not have an applicable model.

Reply:
I'm mostly in agreement:
- Yeah, instead of having constant photoelectric cross section below the energy where the model is applicable we could have a separate photon absorption-only process, but I don't think we want the same for particles with energy loss processes, which we want to track down to zero.
- Agreed, if it isn't applicable we shouldn't interact (and yes, our interactors for energy loss processes already handle this internally)
- That could be useful (and better than the Geant4 way of setting the first value to zero if the xs should be zero below the grid)

We can certainly import them, and these limits should work for the models of energy loss processes, but I don't think we should be using these as the applicabilities for the other models.